lib/Mojolicious/Guides/Tutorial.pod

```diff
@@ -0,0 +1,938 @@
+
+  use Mojolicious::Lite -signatures;
+
+  # Modify the received JSON document and return it
+  put '/reverse' => sub ($c) {
+    my $hash = $c->req->json;
+    $hash->{message} = reverse $hash->{message};
+    $c->render(json => $hash);
+  };
+
+  app->start;
+
+You can send JSON documents from the command line with L<Mojolicious::Command::get>.
+
+  $ ./myapp.pl get -M PUT -c '{"message":"Hello Mojo!"}' /reverse
+
+=head2 Built-in C<exception> and C<not_found> pages
+
+During development you will encounter these pages whenever you make a mistake, they are gorgeous and contain a lot of
+valuable information that will aid you in debugging your application.
+
+  use Mojolicious::Lite -signatures;
+
+  # Not found (404)
+  get '/missing' => sub ($c) {
+    $c->render(template => 'does_not_exist');
+  };
+
+  # Exception (500)
+  get '/dies' => sub { die 'Intentional error' };
+
+  app->start;
+
+You can even use CSS selectors with L<Mojolicious::Command::get> to extract only the information you're actually
+interested in.
+
+  $ ./myapp.pl get /dies '#error'
+
+And don't worry about revealing too much information on these pages, they are only available during development, and
+will be replaced automatically with pages that don't reveal any sensitive information in a production environment.
+
+=head2 Route names
+
+All routes can have a name associated with them, this allows automatic template detection and backreferencing with
+L<Mojolicious::Controller/"url_for">, on which many methods and helpers like
+L<Mojolicious::Plugin::TagHelpers/"link_to"> rely.
+
+  use Mojolicious::Lite -signatures;
+
+  # Render the template "index.html.ep"
+  get '/' => sub ($c) {
+    $c->render;
+  } => 'index';
+
+  # Render the template "hello.html.ep"
+  get '/hello';
+
+  app->start;
+  __DATA__
+
+  @@ index.html.ep
+  <%= link_to Hello  => 'hello' %>.
+  <%= link_to Reload => 'index' %>.
+
+  @@ hello.html.ep
+  Hello World!
+
+Nameless routes get an automatically generated one assigned that is simply equal to the route itself without non-word
+characters.
+
+=head2 Layouts
+
+Templates can have layouts too, you just select one with the helper L<Mojolicious::Plugin::DefaultHelpers/"layout"> and
+place the result of the current template with the helper L<Mojolicious::Plugin::DefaultHelpers/"content">.
+
+  use Mojolicious::Lite;
+
+  get '/with_layout';
+
+  app->start;
+  __DATA__
+
+  @@ with_layout.html.ep
+  % title 'Green';
+  % layout 'green';
+  Hello World!
+
+  @@ layouts/green.html.ep
+  <!DOCTYPE html>
+  <html>
+    <head><title><%= title %></title></head>
+    <body><%= content %></body>
+  </html>
+
+The stash or helpers like L<Mojolicious::Plugin::DefaultHelpers/"title"> can be used to pass additional data to the
+layout.
+
+=head2 Blocks
+
+Template blocks can be used like normal Perl functions and are always delimited by the C<begin> and C<end> keywords,
+they are the foundation for many helpers.
```

Comment:
In what way are "Blocks" the foundation for many helpers?  I'm not certain I've ever used a Block in a helper...  I feel like I'm missing out!